packages/preferences-api/src/browser/preference-proxy.ts

```diff
@@ -46,8 +46,12 @@ export function createPreferenceProxy<T extends Configuration>(preferences: Pref
                     });
                 }
             } else {
-                // TODO If it's deleted, fire the default preference
-                onPreferenceChangedEmitter.fire(e);
+                /* Deleted preference, fire the default preference */
+                onPreferenceChangedEmitter.fire({
+                    preferenceName: e.preferenceName,
+                    newValue: configuration[e.preferenceName],
```

Comment:
What if e.preferenceName is not in configuration, should not we use `schema.properties[p].default`?

Reply:
Cannot `configuration[e.preferenceName]` be different? Should not it win over `schema.properties[p].default`?

Reply:
I was wondering about that also, for now maybe we can use the configuration preference and fallback to the schema, like it was intended to at first by @svenefftinge I believe?

Reply:
I think it was introduced by me and I just did not think about using the default value from a scheme. Could you check existing clients and if configurations are redundant then please get rid of them? If not then we can go with configuration defaults win over scheme defaults.

Reply:
Yea I can try using only the schema if that simplifies things, let me try it.

Reply:
@akosyakov Let me know what you think of the latest commit, I've removed the configurations and came across some issues with the proxy object, so I've change `preference-proxy` a bit. If anything the branch is on this repo, so feel free to change it directly if everything else is fine with the PR.